app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/uploadsupportingdocs/UploadSupportingEvidenceController.scala

```diff
@@ -309,26 +268,26 @@ class UploadSupportingEvidenceController @Inject() (
 
   def uploadSupportingEvidence(): Action[AnyContent] =
     authenticatedActionWithSessionData.async { implicit request =>
-      withUploadSupportingEvidenceAnswers(request) { (draftReturnId, cgtRef, _, _, _) =>
-        upscanService.initiate(DraftReturnId(draftReturnId.toString), cgtRef, LocalDateTime.now()).value.map {
-          case Left(error) =>
-            error.value match {
-              case Left(_) => errorHandler.errorResult()
-              case Right(MaxUploads) =>
-                Redirect(routes.UploadSupportingEvidenceController.checkYourAnswers())
-              case Right(_) => errorHandler.errorResult()
-            }
-          case Right(upscanInitiateResponse) =>
-            upscanInitiateResponse match {
-              case UpscanInitiateSuccess(reference) =>
+      withUploadSupportingEvidenceAnswers(request) { (draftReturnId, cgtRef, _, _, answers) =>
+        if (answers.fold(_.evidences, _.evidences).length >= maxUploads)
```

Comment:
I moved the check here so that we're counting what's in the draft return rather than looking to see what the BE says.